lib/libc/minimal/source/stdlib/malloc.c

```diff
@@ -4,6 +4,8 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+#ifdef CONFIG_MINIMAL_LIBC_MALLOC
```

Comment:
Instead of #if'ing out a whole file, better to just eliminate this from the build (e.g. with a zephyr_library_sources_if_kconfig() in the cmake configuration).

Reply:
Agreed, this can be done in cmake

Reply:
Is this still possible(/desirable) if I switch to using multiple `#ifdef`s in one source file?

Reply:
no, if you need this file to be compiled due to multiple ifdefs I don't think CMake supports that and you'd have to go back to your current approach

This is a REALLY trivial thing to get hung up on guys